src/main/java/com/ibm/watson/modelmesh/ModelMeshApi.java

```diff
@@ -804,8 +807,8 @@ private void processPayload(ByteBuf data, String payloadId, String modelId, Stri
                 Payload payload = null;
                 try {
                     assert payloadProcessor != null;
-                    if (!takeOwnership) {
-                        data.retain();
+                    if (!takeOwnership && data != null) {
+                        ReferenceCountUtil.release(data);
```

Comment:
this seems to do the opposite now, is that intended?

Reply:
:facepalm: good catch, thanks @ckadner. Now fixed